From 2df200ab234a86836a8879a05a8007d6b884eb14 Mon Sep 17 00:00:00 2001
From: Chuhong Yuan <hslester96@gmail.com>
Date: Sun, 10 Nov 2019 07:28:15 +0100
Subject: [PATCH] media: si470x-i2c: add missed operations in remove
Git-commit: 2df200ab234a86836a8879a05a8007d6b884eb14
Patch-mainline: v5.5-rc1
References: git-fixes

The driver misses calling v4l2_ctrl_handler_free and
v4l2_device_unregister in remove like what is done in probe failure.
Add the calls to fix it.

Signed-off-by: Chuhong Yuan <hslester96@gmail.com>
Signed-off-by: Hans Verkuil <hverkuil-cisco@xs4all.nl>
Signed-off-by: Mauro Carvalho Chehab <mchehab@kernel.org>
Acked-by: Takashi Iwai <tiwai@suse.de>

---
 drivers/media/radio/si470x/radio-si470x-i2c.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/media/radio/si470x/radio-si470x-i2c.c
+++ b/drivers/media/radio/si470x/radio-si470x-i2c.c
@@ -473,6 +473,8 @@ static int si470x_i2c_suspend(struct dev
 	if (si470x_set_register(radio, POWERCFG) < 0)
 		return -EIO;
 
+	v4l2_ctrl_handler_free(&radio->hdl);
+	v4l2_device_unregister(&radio->v4l2_dev);
 	return 0;
 }
 
